shotgun_api3/shotgun.py

```diff
@@ -1473,20 +1494,33 @@ def update_project_last_accessed(self, project, user=None):
         record = self._call_rpc("update_project_last_accessed_by_current_user", params)
         result = self._parse_records(record)[0]
 
-
-    def _get_session_token(self):
-        """Hack to authenticate in order to download protected content
-        like Attachments
+    def generate_session_token(self):
+        """
+        Generate a session token that can be used to authenticate the currently 
+        logged in user or script.
+        
+        :returns: String containing a session token
         """
-        if self.config.session_token:
-            return self.config.session_token
-
```

Comment:
Also, why not return the cached session if it is set?  The reason for the difference between this and the protected _get_session_token is not clear.